OpenEdXMobile/src/main/java/org/edx/mobile/view/AccountFragment.kt

```diff
@@ -115,17 +119,19 @@ class AccountFragment : BaseFragment() {
             BuildConfig.VERSION_NAME, config.environmentDisplayName
         )
 
-        config.deleteAccountUrl?.let { deleteAccountUrl ->
-            binding.containerDeleteAccount.visibility = View.VISIBLE
-            binding.btnDeleteAccount.setOnClickListener {
-                environment.router.showAuthenticatedWebViewActivity(
-                    this.requireContext(),
-                    deleteAccountUrl, getString(R.string.title_delete_my_account), false
-                )
-                trackEvent(
-                    Analytics.Events.DELETE_ACCOUNT_CLICKED,
-                    Analytics.Values.DELETE_ACCOUNT_CLICKED
-                )
+        if (loginPrefs.isUserLoggedIn) {
+            config.deleteAccountUrl?.let { deleteAccountUrl ->
```

Comment:
we are checking the user logged-In on line:107, we can move this under that check instead separately checking on different places.